File: wagerproof-ios-native/Wagerproof/Features/Agents/Components/AgentPickPayloadAuditWidget.swift
import SwiftUI
import UIKit
import WagerproofDesign
import WagerproofModels
import WagerproofStores

/// Native port of `components/agents/AgentPickPayloadAuditWidget.tsx`. The
/// "terminal" debug surface that dumps the raw audit payload (game inputs,
/// personality inputs, model response, leaned metrics) for a single pick.
///
/// Driven by `AgentPickAuditStore.payload`. Includes a copy-to-clipboard
/// button so developers / debugging users can paste the dump elsewhere.
struct AgentPickPayloadAuditWidget: View {
    let pick: AgentPick
    let payload: AgentPickAuditPayload

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // Header — terminal-style pick id label + full-trace export
            HStack {
                Text("terminal://pick-audit/\(pick.id.prefix(8))")
                    .font(.system(size: 12, design: .monospaced))
                    .foregroundStyle(Color(hex: 0x9FB3AD))
                Spacer()
                // One-tap export of the ENTIRE trace (pick + ai_decision_trace +
                // ai_audit_payload) so it can be pasted into a bug report / chat.
                Button {
                    UIPasteboard.general.string = payload.fullTraceJSON
                } label: {
                    HStack(spacing: 4) {
                        Image(systemName: "doc.on.doc.fill")
                            .font(.system(size: 11, weight: .semibold))
                        Text("Copy Full Trace")
                            .font(.system(size: 11, weight: .heavy, design: .monospaced))
                            .tracking(0.5)
                    }
                    .foregroundStyle(Color(hex: 0x0B1010))
                    .padding(.horizontal, 10)
                    .padding(.vertical, 5)
                    .background(Capsule().fill(Color(hex: 0x26DF85)))
                }
                .buttonStyle(.plain)
                .accessibilityLabel("Copy full decision trace")
            }

            // Matchup banner row
            HStack(alignment: .firstTextBaseline, spacing: 8) {
                Text("›")
                    .font(.system(size: 14, weight: .bold, design: .monospaced))
                    .foregroundStyle(Color(hex: 0x00E676))
                Text("\(pick.matchup) | \(pick.pickSelection)")
                    .font(.system(size: 14, weight: .semibold, design: .monospaced))
                    .foregroundStyle(Color(hex: 0x26DF85))
            }

            section(title: "LEANED METRICS") {
                if payload.leanedMetrics.isEmpty {
                    Text("No explicit leaned metrics were returned for this pick.")
                        .font(.system(size: 13, design: .monospaced))
                        .foregroundStyle(Color.appTextSecondary)
                } else {
                    VStack(alignment: .leading, spacing: 10) {
                        ForEach(payload.leanedMetrics) { metric in
                            VStack(alignment: .leading, spacing: 3) {
                                Text("\(metric.metricKey) = \(metric.metricValue)")
                                    .font(.system(size: 13, weight: .semibold, design: .monospaced))
                                    .foregroundStyle(Color(hex: 0x26DF85))
                                Text(metric.whyItMattered)
                                    .font(.system(size: 12, design: .monospaced))
                                    .foregroundStyle(Color.appTextSecondary)
                                Text("trait: \(metric.personalityTrait)")
                                    .font(.system(size: 11, design: .monospaced))
                                    .foregroundStyle(Color.appTextSecondary.opacity(0.7))
                            }
                        }
                    }
                }
            }

            section(title: "WHY THIS PICK") {
                Text(payload.rationaleText)
                    .font(.system(size: 13, design: .monospaced))
                    .foregroundStyle(Color.appTextSecondary)
                    .fixedSize(horizontal: false, vertical: true)
            }

            section(title: "PERSONALITY ALIGNMENT") {
                Text(payload.personalityAlignmentText)
                    .font(.system(size: 13, design: .monospaced))
                    .foregroundStyle(Color.appTextSecondary)
                    .fixedSize(horizontal: false, vertical: true)
            }

            payloadSection(title: "MODEL INPUT GAME PAYLOAD", json: payload.modelInputGameJSON)
            if !payload.payloadIsFormatted {
                Text("Note: This appears to be a legacy raw snapshot. New picks store the exact formatted model input payload.")
                    .font(.system(size: 11, design: .monospaced))
                    .foregroundStyle(Color(hex: 0xF59E0B))
            }
            payloadSection(title: "AGENT PERSONALITY PAYLOAD", json: payload.modelInputPersonalityJSON)
            payloadSection(title: "AGENT RESPONSE PAYLOAD", json: payload.modelResponseJSON)
            if !payload.toolTrace.isEmpty {
                toolCallsSection
            }
        }
        .padding(14)
        .background(
            RoundedRectangle(cornerRadius: 12, style: .continuous)
                .fill(Color(hex: 0x050909))
        )
        .overlay(
            RoundedRectangle(cornerRadius: 12, style: .continuous)
                .strokeBorder(Color(hex: 0x00E676).opacity(0.25), lineWidth: 1)
        )
    }

    /// V3 generation loop's tool calls (seq 0 = the slate the model saw), each
    /// with its raw result excerpt so the audit shows what the agent fetched.
    private var toolCallsSection: some View {
        VStack(alignment: .leading, spacing: 6) {
            HStack {
                Text("TOOL CALLS")
                    .font(.system(size: 11, weight: .heavy, design: .monospaced))
                    .tracking(1.2)
                    .foregroundStyle(Color(hex: 0x9FB3AD))
                Spacer()
                Button {
                    UIPasteboard.general.string = payload.toolTraceJSON
                } label: {
                    HStack(spacing: 3) {
                        Image(systemName: "doc.on.doc")
                            .font(.system(size: 10, weight: .semibold))
                        Text("Copy")
                            .font(.system(size: 10, weight: .heavy, design: .monospaced))
                            .tracking(0.5)
                    }
                    .foregroundStyle(Color(hex: 0x26DF85))
                }
                .buttonStyle(.plain)
                .accessibilityLabel("Copy tool calls")
            }
            VStack(alignment: .leading, spacing: 10) {
                ForEach(payload.toolTrace) { entry in
                    VStack(alignment: .leading, spacing: 3) {
                        Text("\(entry.name) · \(entry.ms)ms · \(entry.ok ? "ok" : "FAIL")")
                            .font(.system(size: 13, weight: .semibold, design: .monospaced))
                            .foregroundStyle(entry.ok ? Color(hex: 0x26DF85) : Color(hex: 0xF59E0B))
                        if !entry.resultExcerpt.isEmpty {
                            ScrollView(.horizontal, showsIndicators: false) {
                                Text(entry.resultExcerpt)
                                    .font(.system(size: 11, design: .monospaced))
                                    .foregroundStyle(Color(hex: 0x8CA89B))
                                    .padding(10)
                                    .frame(maxWidth: .infinity, alignment: .leading)
                                    .background(
                                        RoundedRectangle(cornerRadius: 8, style: .continuous)
                                            .fill(Color.black.opacity(0.4))
                                    )
                            }
                        }
                    }
                }
            }
        }
        .padding(.top, 8)
    }

    @ViewBuilder
    private func section<Content: View>(title: String, @ViewBuilder content: () -> Content) -> some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(title)
                .font(.system(size: 11, weight: .heavy, design: .monospaced))
                .tracking(1.2)
                .foregroundStyle(Color(hex: 0x9FB3AD))
            content()
        }
        .padding(.top, 8)
    }

    private func payloadSection(title: String, json: String) -> some View {
        VStack(alignment: .leading, spacing: 6) {
            HStack {
                Text(title)
                    .font(.system(size: 11, weight: .heavy, design: .monospaced))
                    .tracking(1.2)
                    .foregroundStyle(Color(hex: 0x9FB3AD))
                Spacer()
                Button {
                    UIPasteboard.general.string = json
                } label: {
                    HStack(spacing: 3) {
                        Image(systemName: "doc.on.doc")
                            .font(.system(size: 10, weight: .semibold))
                        Text("Copy")
                            .font(.system(size: 10, weight: .heavy, design: .monospaced))
                            .tracking(0.5)
                    }
                    .foregroundStyle(Color(hex: 0x26DF85))
                }
                .buttonStyle(.plain)
                .accessibilityLabel("Copy \(title)")
            }
            ScrollView(.horizontal, showsIndicators: false) {
                Text(json)
                    .font(.system(size: 11, design: .monospaced))
                    .foregroundStyle(Color(hex: 0x8CA89B))
                    .padding(10)
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .background(
                        RoundedRectangle(cornerRadius: 8, style: .continuous)
                            .fill(Color.black.opacity(0.4))
                    )
            }
        }
        .padding(.top, 8)
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/agents/components/AgentPickPayloadAuditWidget.kt
package com.wagerproof.app.features.agents.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalClipboardManager
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.app.features.agents.agentSymbol
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.AgentPick
import com.wagerproof.core.models.AgentPickAuditPayload

// =====================================================================
// AgentPickPayloadAuditWidget — terminal debug surface dumping a pick's raw
// audit payload (game inputs, personality, model response, leaned metrics,
// tool trace). 5 copy-to-clipboard buttons. Port of iOS
// AgentPickPayloadAuditWidget.swift.
// =====================================================================

private val auditGreen = AppColors.brandGreenBright          // 0x00E676
private val auditGreenMid = Color(0xFF26DF85)
private val auditGreenDim = Color(0xFF8CA89B)
private val auditHeaderColor = Color(0xFF9FB3AD)
private val auditInk = Color(0xFF0B1010)
private val auditAmber = Color(0xFFF59E0B)
private val auditCardBg = Color(0xFF050909)

@Composable
fun AgentPickPayloadAuditWidget(
    pick: AgentPick,
    payload: AgentPickAuditPayload,
    modifier: Modifier = Modifier,
) {
    val clipboard = LocalClipboardManager.current
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(auditCardBg)
            .border(1.dp, auditGreen.copy(alpha = 0.25f), RoundedCornerShape(12.dp))
            .padding(14.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        // Header + full-trace export.
        Row(verticalAlignment = Alignment.CenterVertically, modifier = Modifier.fillMaxWidth()) {
            Text(
                "terminal://pick-audit/${pick.id.take(8)}",
                color = auditHeaderColor, fontSize = 12.sp, fontFamily = FontFamily.Monospace,
            )
            Spacer(Modifier.weight(1f))
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(4.dp),
                modifier = Modifier
                    .clip(CircleShape)
                    .background(auditGreenMid)
                    .clickable { clipboard.setText(AnnotatedString(payload.fullTraceJSON)) }
                    .padding(horizontal = 10.dp, vertical = 5.dp),
            ) {
                Icon(agentSymbol("doc.on.doc.fill"), null, tint = auditInk, modifier = Modifier.size(11.dp))
                Text("Copy Full Trace", color = auditInk, fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, fontFamily = FontFamily.Monospace)
            }
        }

        // Matchup banner.
        Row(verticalAlignment = Alignment.Top, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            Text("›", color = auditGreen, fontSize = 14.sp, fontWeight = FontWeight.Bold, fontFamily = FontFamily.Monospace)
            Text(
                "${pick.matchup} | ${pick.pickSelection}",
                color = auditGreenMid, fontSize = 14.sp, fontWeight = FontWeight.SemiBold, fontFamily = FontFamily.Monospace,
            )
        }

        // LEANED METRICS.
        AuditSection("LEANED METRICS") {
            if (payload.leanedMetrics.isEmpty()) {
                Text("No explicit leaned metrics were returned for this pick.", color = AppColors.appTextSecondary, fontSize = 13.sp, fontFamily = FontFamily.Monospace)
            } else {
                Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                    payload.leanedMetrics.forEach { metric ->
                        Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
                            Text("${metric.metricKey} = ${metric.metricValue}", color = auditGreenMid, fontSize = 13.sp, fontWeight = FontWeight.SemiBold, fontFamily = FontFamily.Monospace)
                            Text(metric.whyItMattered, color = AppColors.appTextSecondary, fontSize = 12.sp, fontFamily = FontFamily.Monospace)
                            Text("trait: ${metric.personalityTrait}", color = AppColors.appTextSecondary.copy(alpha = 0.7f), fontSize = 11.sp, fontFamily = FontFamily.Monospace)
                        }
                    }
                }
            }
        }

        AuditSection("WHY THIS PICK") {
            Text(payload.rationaleText, color = AppColors.appTextSecondary, fontSize = 13.sp, fontFamily = FontFamily.Monospace)
        }
        AuditSection("PERSONALITY ALIGNMENT") {
            Text(payload.personalityAlignmentText, color = AppColors.appTextSecondary, fontSize = 13.sp, fontFamily = FontFamily.Monospace)
        }

        PayloadSection("MODEL INPUT GAME PAYLOAD", payload.modelInputGameJSON, clipboard)
        if (!payload.payloadIsFormatted) {
            Text(
                "Note: This appears to be a legacy raw snapshot. New picks store the exact formatted model input payload.",
                color = auditAmber, fontSize = 11.sp, fontFamily = FontFamily.Monospace,
            )
        }
        PayloadSection("AGENT PERSONALITY PAYLOAD", payload.modelInputPersonalityJSON, clipboard)
        PayloadSection("AGENT RESPONSE PAYLOAD", payload.modelResponseJSON, clipboard)

        if (payload.toolTrace.isNotEmpty()) {
            Column(Modifier.padding(top = 8.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                Row(verticalAlignment = Alignment.CenterVertically, modifier = Modifier.fillMaxWidth()) {
                    SectionTitle("TOOL CALLS")
                    Spacer(Modifier.weight(1f))
                    CopyButton { clipboard.setText(AnnotatedString(payload.toolTraceJSON)) }
                }
                Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                    payload.toolTrace.forEach { entry ->
                        Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
                            Text(
                                "${entry.name} · ${entry.ms}ms · ${if (entry.ok) "ok" else "FAIL"}",
                                color = if (entry.ok) auditGreenMid else auditAmber,
                                fontSize = 13.sp, fontWeight = FontWeight.SemiBold, fontFamily = FontFamily.Monospace,
                            )
                            if (entry.resultExcerpt.isNotEmpty()) {
                                JsonBlock(entry.resultExcerpt)
                            }
                        }
                    }
                }
            }
        }
    }
}

@Composable
private fun SectionTitle(title: String) {
    Text(title, color = auditHeaderColor, fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, fontFamily = FontFamily.Monospace)
}

@Composable
private fun AuditSection(title: String, content: @Composable () -> Unit) {
    Column(Modifier.padding(top = 8.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
        SectionTitle(title)
        content()
    }
}

@Composable
private fun PayloadSection(
    title: String,
    json: String,
    clipboard: androidx.compose.ui.platform.ClipboardManager,
) {
    Column(Modifier.padding(top = 8.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
        Row(verticalAlignment = Alignment.CenterVertically, modifier = Modifier.fillMaxWidth()) {
            SectionTitle(title)
            Spacer(Modifier.weight(1f))
            CopyButton { clipboard.setText(AnnotatedString(json)) }
        }
        JsonBlock(json)
    }
}

@Composable
private fun CopyButton(onClick: () -> Unit) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(3.dp),
        modifier = Modifier.clickable(onClick = onClick),
    ) {
        Icon(agentSymbol("doc.on.doc"), null, tint = auditGreenMid, modifier = Modifier.size(10.dp))
        Text("Copy", color = auditGreenMid, fontSize = 10.sp, fontWeight = FontWeight.ExtraBold, fontFamily = FontFamily.Monospace)
    }
}

@Composable
private fun JsonBlock(text: String) {
    Box(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(8.dp))
            .background(Color.Black.copy(alpha = 0.4f))
            .horizontalScroll(rememberScrollState())
            .padding(10.dp),
    ) {
        Text(text, color = auditGreenDim, fontSize = 11.sp, fontFamily = FontFamily.Monospace)
    }
}
